test/e2e/plain_provisioner_test.go

```diff
@@ -544,6 +544,134 @@ var _ = Describe("plain provisioner bundleinstance", func() {
 		})
 	})
 
+	When("a BundleInstance is dependent on another BundleInstance", func() {
+		var (
+			ctx             context.Context
+			dependentBundle *rukpakv1alpha1.Bundle
+			dependentBI     *rukpakv1alpha1.BundleInstance
+		)
+		BeforeEach(func() {
+			ctx = context.Background()
+
+			By("creating the testing dependent Bundle resource")
+			dependentBundle = &rukpakv1alpha1.Bundle{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "e2e-bundle-dependent-",
+				},
+				Spec: rukpakv1alpha1.BundleSpec{
+					ProvisionerClassName: plainProvisionerID,
+					Source: rukpakv1alpha1.BundleSource{
+						Type: rukpakv1alpha1.SourceTypeImage,
+						Image: &rukpakv1alpha1.ImageSource{
+							Ref: "testdata/bundles/plain-v0:dependent",
+						},
+					},
+				},
+			}
+			err := c.Create(ctx, dependentBundle)
+			Expect(err).To(BeNil())
+
+			By("creating the testing dependent BundleInstance resource")
+			dependentBI = &rukpakv1alpha1.BundleInstance{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "e2e-bi-dependent-",
+				},
+				Spec: rukpakv1alpha1.BundleInstanceSpec{
+					ProvisionerClassName: plainProvisionerID,
+					BundleName:           dependentBundle.GetName(),
+				},
+			}
+			err = c.Create(ctx, dependentBI)
+			Expect(err).To(BeNil())
+		})
+		AfterEach(func() {
+			By("deleting the testing dependent Bundle resource")
+			Expect(client.IgnoreNotFound(c.Delete(ctx, dependentBundle))).To(BeNil())
+
+			By("deleting the testing dependent BundleInstance resource")
+			Expect(client.IgnoreNotFound(c.Delete(ctx, dependentBI))).To(BeNil())
+
+		})
+		When("the providing BundleInstance does not exist", func() {
+			It("should eventually project a failed installation for the dependent BundleInstance", func() {
+				Eventually(func() (*metav1.Condition, error) {
+					if err := c.Get(ctx, client.ObjectKeyFromObject(dependentBI), dependentBI); err != nil {
+						return nil, err
+					}
+					return meta.FindStatusCondition(dependentBI.Status.Conditions, rukpakv1alpha1.TypeInstalled), nil
+				}).Should(And(
+					Not(BeNil()),
+					WithTransform(func(c *metav1.Condition) string { return c.Type }, Equal(rukpakv1alpha1.TypeInstalled)),
+					WithTransform(func(c *metav1.Condition) metav1.ConditionStatus { return c.Status }, Equal(metav1.ConditionFalse)),
+					WithTransform(func(c *metav1.Condition) string { return c.Reason }, Equal(rukpakv1alpha1.ReasonInstallFailed)),
+					WithTransform(func(c *metav1.Condition) string { return c.Message }, ContainSubstring(`uninstall failed`)),
```

Comment:
This is a strange message -- should it be `install failed`? If we aren't uninstalling in this context?

Reply:
Yep - It looks like the failure we get back isn't always deterministic which is slightly annoying. I ended up changing this to check the error message is complaining about the OG CRD being missing on cluster, which should be a bit more readable and easier to reason about.